frame/lending/src/helpers.rs

```diff
@@ -243,6 +243,11 @@ impl<T: Config> Pallet<T> {
 			Error::<T>::MarketIsClosing
 		);
 
+		ensure!(
+			T::Vault::is_vault_balanced(&market.borrow_asset_vault, &market_account)?,
+			Error::<T>::CannotBorrowFromMarketWithUnbalancedVault
+		);
+
```

Comment:
can you merge these two ensures into one function, so we aren't querying available_funds back to back?

Reply:
I think a better way of doing this check is creating a function e.g. `can_borrow_from_vault`. This function can contain the logic for querying `available_funds` and return:
1. Error::<T>::MarketIsClosing if `available_funds` returns FundsAvailability::MustLiquidate
2. Error::<T>::CannotBorrowFromMarketWithUnbalancedVault if if `available_funds` returns FundsAvailability::Withdrawable(balance)
3. Ok() otherwise

Reply:
I think the logic of `is_vault_balanced` is better contained within the `lending` pallet rather than the `vault` pallet

Reply:
Done